src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1278,25 +1278,43 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
           return glnx_throw (error, "No overrides for package '%s'", name_or_nevra);
         }
 
-      changed = TRUE;
-    }
-
-  if (self->override_replace_local_pkgs)
-    {
-      g_autoptr(GPtrArray) pkgs = NULL;
-      if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
-                                   cancellable, error))
-        return FALSE;
-
-      if (pkgs->len > 0)
+      if (self->override_replace_local_pkgs)
         {
-          g_ptr_array_add (pkgs, NULL);
-          if (!rpmostree_origin_add_overrides (origin, (char**)pkgs->pdata,
-                                               RPMOSTREE_ORIGIN_OVERRIDE_REPLACE_LOCAL,
-                                               error))
+          g_autoptr(GPtrArray) pkgs = NULL;
+          if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
+                                       cancellable, error))
             return FALSE;
-          changed = TRUE;
-        }
+
+          for (guint i = 0; i < pkgs->len; i++)
+            {
+
+              const char *pkg = static_cast<const char *>(g_ptr_array_index(pkgs,i));
+              g_autofree char *name = NULL;
+              g_autofree char *sha256 = NULL;
+
+              if(!rpmostree_decompose_sha256_nevra(&pkg, &sha256, error))
+                return FALSE;
+
+              if(!rpmostree_decompose_nevra(pkg, &name, NULL, NULL, NULL, NULL, error))
+                return FALSE;
+
+              auto nevra = static_cast<const char *>(g_hash_table_lookup (name_to_nevra, name));
+              if (nevra == NULL)
+                nevra = pkg;
+              if (rpmostree_origin_remove_override (origin, nevra,
+                                                    RPMOSTREE_ORIGIN_OVERRIDE_REPLACE_LOCAL))
+                continue; /* override found; move on to the next one */
```

Comment:
That make sense, because I think nevra cannot be NULL in `rpmostree_origin_remove_override`, and passing a random value is senseless.

Reply:
I was doubting about the "Allow" word in the commit message because it's something that *will* happen in a specific context, is not an option. But I think is accurate enough because it let's you do something you couldn't do before. So I'll keep it.